### ISSUE TITLE
Adjust probability of n_sources in detection encoder at test time

### ISSUE THREAD

**Issue by ismael-mendoza** [prob-ml/bliss#509]
By adjusting the output of the detection encoder  

```
beta_0^{\star} = beta_0 + log (pi / (1 - pi)) - log ( pi^{\tilde}  / (1 - pi^{\tilde}) 
```

used for logistic regression but should work with softmax too. `pi` is correct prior (test data) and `pi^{\tilde}` is the training set prior.

**Comment by dereklhansen** [comment]
Here's the full derivation of how to do it in BLISS:

![image](https://user-images.githubusercontent.com/12701492/160933857-d7512003-77f0-4f1b-80f5-2c3de18c8caa.png)
